src/librustdoc/clean/mod.rs

```diff
@@ -1582,12 +1582,12 @@ fn normalize<'tcx>(cx: &mut DocContext<'tcx>, ty: Ty<'_>) -> Option<Ty<'tcx>> {
 }
 
 pub(crate) fn clean_middle_ty<'tcx>(
-    this: Ty<'tcx>,
+    ty: Ty<'tcx>,
```

Comment:
Intentionally renaming `this` -> `ty` so that the normalized `let ty = ...` below shadows the unnormalized type, so this doesn't happen again.